### ISSUE TITLE
fatal error, then panic: runtime error

### ISSUE THREAD

**Issue by skyfaller** [ashishb/wp2hugo#15]
I tried running this on my Wordpress XML export and I got an error message complaining about a bad date:

```
> ./bin/wp2hugo --source ~/Downloads/wrongsideofdawn.WordPress.2024-04-29.xml
{"level":"debug","time":1714428061,"message":"Source: /home/nelson/Downloads/wrongsideofdawn.WordPress.2024-04-29.xml"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2009/06/wsodcover_front2-500-1.png","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/03/down-1.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/03/apple-touch-icon-1.png","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/04/saint-1.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/04/01-I-Get-Scared-2006-12-19-1.mp3","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/04/stevie_wonder-1.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/04/keith-1.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/bass-tabs-Wrong-Side-of-Dawn-Stay-Awake-1.zip","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/bass-tabs-in-TuxGuitar-format-WSD-1.zip","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Running-Scared-1.pdf","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Out-of-Time-1.pdf","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Flight-II-1.pdf","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Break-Free-1.pdf","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Last_Warning-1.pdf","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Contained-1.pdf","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-My-Private-Asylum-1.pdf","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Grinders-Tale-1.pdf","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Where-Is-Bobby-McGee-1.pdf","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Flight-3-1.pdf","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Tuxguitar-1.png","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/4c29bdab-9788-4de4-affc-c9406de2c492-1.jpeg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/08/poolside3_500-1.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/09/Running-Out-of-Time-version-1-no-1.mp3","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/09/JohnnyRzeznik-1.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/09/brian-playing-1.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/09/davidgilmoure-1.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2011/12/Rock-Band-playing-11-1.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2011/12/Rock-Band-playing-2-1.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2013/08/IMG_1583-1.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2013/08/09-Classical-Grinders-Tale-1.m4a","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2015/06/Photo-on-6-24-15-at-11.53-AM-1.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2015/06/FirstFridayJuly3-1.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2015/08/WSoD-MainLogo-colors1-1.png-again-1.png","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2015/08/cropped-WSoD-MainLogo-colors1-1.png-again-1.png","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/3384355592_2b13feac62_o_mini.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/cropped-3384355592_2b13feac62_o_mini.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/3383557883_176804dbe6_o_mini.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/cropped-3383557883_176804dbe6_o_mini.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/3383543549_651c6ec034_o_mini.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/cropped-3383543549_651c6ec034_o_mini.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/5641172105_0412909256_o-1_mini.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/cropped-5641172105_0412909256_o-1_mini.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/wsd-logogo.png","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/cropped-wsd-logogo.png","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/cropped-wsd-logogo-1.png","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2013/08/IMG_1583-1_mini.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2015/06/FirstFridayJuly3-1-768x994-1.jpg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2018/08/wsd_logo.svg","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2021/09/flight3-nosound.webm","time":1714428061,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2021/09/flight3-x264.mp4","time":1714428061,"message":"Attachment URL"}
{"level":"fatal","time":1714428061,"message":"Error: error getting common fields: error parsing last modified date: error parsing time: parsing time \"0000-00-00 00:00:00\": month out of range"}
```

So I manually deleted every line that had "0000-00-00" in it from my xml file. The relevant lines looked like:
```
<wp:post_date_gmt><![CDATA[0000-00-00 00:00:00]]></wp:post_date_gmt>
<wp:post_modified_gmt><![CDATA[0000-00-00 00:00:00]]></wp:post_modified_gmt>
```

So I figured they didn't contain any important info; these were redundant post dates for a date that was not in GMT, and they happened to be incorrect.

Then I got a new error:

```
> ./bin/wp2hugo --source ~/Downloads/wrongsideofdawn.WordPress.2024-04-29.xml
{"level":"debug","time":1714428580,"message":"Source: /home/nelson/Downloads/wrongsideofdawn.WordPress.2024-04-29.xml"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2009/06/wsodcover_front2-500-1.png","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/03/down-1.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/03/apple-touch-icon-1.png","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/04/saint-1.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/04/01-I-Get-Scared-2006-12-19-1.mp3","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/04/stevie_wonder-1.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/04/keith-1.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/bass-tabs-Wrong-Side-of-Dawn-Stay-Awake-1.zip","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/bass-tabs-in-TuxGuitar-format-WSD-1.zip","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Running-Scared-1.pdf","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Out-of-Time-1.pdf","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Flight-II-1.pdf","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Break-Free-1.pdf","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Last_Warning-1.pdf","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Contained-1.pdf","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-My-Private-Asylum-1.pdf","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Grinders-Tale-1.pdf","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Where-Is-Bobby-McGee-1.pdf","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Wrong-Side-of-Dawn-Flight-3-1.pdf","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/Tuxguitar-1.png","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/06/4c29bdab-9788-4de4-affc-c9406de2c492-1.jpeg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/08/poolside3_500-1.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/09/Running-Out-of-Time-version-1-no-1.mp3","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/09/JohnnyRzeznik-1.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/09/brian-playing-1.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2010/09/davidgilmoure-1.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2011/12/Rock-Band-playing-11-1.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2011/12/Rock-Band-playing-2-1.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2013/08/IMG_1583-1.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2013/08/09-Classical-Grinders-Tale-1.m4a","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2015/06/Photo-on-6-24-15-at-11.53-AM-1.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2015/06/FirstFridayJuly3-1.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2015/08/WSoD-MainLogo-colors1-1.png-again-1.png","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2015/08/cropped-WSoD-MainLogo-colors1-1.png-again-1.png","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/3384355592_2b13feac62_o_mini.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/cropped-3384355592_2b13feac62_o_mini.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/3383557883_176804dbe6_o_mini.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/cropped-3383557883_176804dbe6_o_mini.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/3383543549_651c6ec034_o_mini.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/cropped-3383543549_651c6ec034_o_mini.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/5641172105_0412909256_o-1_mini.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/cropped-5641172105_0412909256_o-1_mini.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/wsd-logogo.png","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/cropped-wsd-logogo.png","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2017/05/cropped-wsd-logogo-1.png","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2013/08/IMG_1583-1_mini.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2015/06/FirstFridayJuly3-1-768x994-1.jpg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2018/08/wsd_logo.svg","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2021/09/flight3-nosound.webm","time":1714428580,"message":"Attachment URL"}
{"level":"debug","attachmentURL":"https://www.wrongsideofdawn.com/wp-content/uploads/2021/09/flight3-x264.mp4","time":1714428580,"message":"Attachment URL"}
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/ashishb/wp2hugo/src/wp2hugo/internal/wpparser.getCommonFields(0xc0012b5100)
	/home/nelson/code/wp2hugo/src/wp2hugo/internal/wpparser/wp_parser_setup.go:259 +0xd79
github.com/ashishb/wp2hugo/src/wp2hugo/internal/wpparser.getPostInfo(0x7ed6a0?)
	/home/nelson/code/wp2hugo/src/wp2hugo/internal/wpparser/wp_parser_setup.go:429 +0x25
github.com/ashishb/wp2hugo/src/wp2hugo/internal/wpparser.(*Parser).getWebsiteInfo(0xc000271ca7?, 0xc0001b9d40)
	/home/nelson/code/wp2hugo/src/wp2hugo/internal/wpparser/wp_parser_setup.go:194 +0x7bf
github.com/ashishb/wp2hugo/src/wp2hugo/internal/wpparser.(*Parser).Parse(0xc000b7bcf8, {0xa95400?, 0xc00008e810?})
	/home/nelson/code/wp2hugo/src/wp2hugo/internal/wpparser/wp_parser_setup.go:152 +0xae
main.getWebsiteInfo({0x7fff671d28fd?, 0x8b56b8?})
	/home/nelson/code/wp2hugo/src/wp2hugo/cmd/wp2hugo/main.go:57 +0x46
main.handle({0x7fff671d28fd, 0x3f})
	/home/nelson/code/wp2hugo/src/wp2hugo/cmd/wp2hugo/main.go:44 +0xba
main.main()
	/home/nelson/code/wp2hugo/src/wp2hugo/cmd/wp2hugo/main.go:36 +0xce
```

I don't know how to interpret this one. What do I do now?

I got the same results with both the binary release from GitHub and building wp2hugo myself.

**Comment by ashishb** [comment]
@skyfaller can you email me the XML (ashishbATashishbDOTnet)?
Even the partial XML of just that one entry would help.

**Comment by skyfaller** [comment]
email sent

**Comment by ashishb** [comment]
Thanks @skyfaller, This has been fixed in release 1.3.
